test/integration/vm_translator_test.rb

```diff
@@ -1,39 +1,14 @@
 require "test_helper"
 
 class VMTranslatorIntegrationTest < Minitest::Test
-  def test_integration_test
-    assembly_code = `bin/vm-translator examples/Push.vm`
-    expected = File.read("test/fixtures/Push.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_simple_add
-    assembly_code = `bin/vm-translator examples/SimpleAdd.vm`
-    expected = File.read("test/fixtures/SimpleAdd.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_simple_eq
-    assembly_code = `bin/vm-translator examples/SimpleEq.vm`
-    expected = File.read("test/fixtures/SimpleEq.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_stack_test
-    assembly_code = `bin/vm-translator examples/StackTest.vm`
-    expected = File.read("test/fixtures/StackTest.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_basic_test
-    assembly_code = `bin/vm-translator examples/BasicTest.vm`
-    expected = File.read("test/fixtures/BasicTest.asm")
-    assert_equal(expected, assembly_code)
-  end
-
-  def test_integration_test_pointer_test
-    assembly_code = `bin/vm-translator examples/PointerTest.vm`
-    expected = File.read("test/fixtures/PointerTest.asm")
-    assert_equal(expected, assembly_code)
+  Dir.glob("examples/**").each do |file_path|
+    file_name = /examples\/(.*).vm/.match(file_path)[1]
```

Comment:
This is a common operation, so happily there’s an easier way to do it. Ruby calls the last part of a file path its “base name”, and provides the [`File.basename`](https://rubydoc.info/stdlib/core/File.basename) method to extract it:

```
>> File.basename("/foo/bar/baz.txt")
=> "baz.txt"
```

What’s particularly useful is that `File.basename` takes an optional second argument which is a suffix to remove from the base name if present:

```
>> File.basename("/foo/bar/baz.txt", ".txt")
=> "baz"
>> File.basename("/foo/bar/qux.html", ".txt")
=> "qux.html"
```

Because this can get a bit repetitive (e.g. “I’ve just globbed for all the `*.txt` files, and now I want to remove all their `.txt` suffixes”), `File.basename` supports a special `".*"` suffix which means “remove any extension” so we don’t have to restate the exact suffix we want to strip:

```
>> File.basename("/foo/bar/baz.txt", ".*")
=> "baz"
>> File.basename("/foo/bar/qux.html", ".*")
=> "qux"
```